torch/csrc/jit/codegen/cuda/kernel.h

```diff
@@ -44,6 +44,13 @@ struct KernelSummary {
   DataType largest_smem_data_type = DataType::Null;
 };
 
+//! Summary of interesting facts for testing / validation
+// about the kernel
+struct DebugKernelSummary {
```

Comment:
I would actually prefer the original, using the existing `KernelSummary` (having a separate structure and scan only duplicates things and adds even more overhead)

Reply:
How about using a separate `KernelSummary` and `DebugKernelSummary` but having a common `KernelIRScanner`? The `DebugKernelIRScanner` is redundant.

Reply:
What does a separate DebugKernelSummay buys us compared to a simple comment for the `war_hazard_syncs` member? 

I'd rather keep things as simple as possible (also, if we change the validation we'd have to delete the DebugKernelSummary anyway so it seems counterproductive to add it in the first place)